edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Interfaces/IMetricsStorage.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    /// <summary>
+    /// Stores metrics for later upload.
+    /// </summary>
+    public interface IMetricsStorage
```

Comment:
why not store in rocksdb? if so, you may reuse ColumnFamilyDbStore or StorageSpaceAwareDbStore.

Reply:
I didn't realize edgeAgent had rocks db. I would like to change it, but I thing that belongs in a different pr. This pr has already grown too big.

Reply:
Sure.  You may need to change some of these signatures.